### ISSUE TITLE
ci: test suite failure — 2026-07-01

### ISSUE THREAD

**Issue by Pavua** [Pavua/Krab-Ear#1822]
## Summary
Daily scheduled test-health check found the Python test suite hangs and never completes (killed by a 900s watchdog timeout after running for 15+ minutes). Swift suite is unaffected and fully green.

## Python suite: HUNG at 80% (not a normal failure)
`PYTHONPATH=$(pwd)/KrabEar python -m pytest KrabEar/tests/ -q --tb=line` progressed to 80% (line ~216/33627 in the raw log) and then stalled. A 900s `timeout` wrapper killed it; pytest-timeout's thread dump shows the real cause:

```
  File ".../KrabEar/tests/test_swift_response_contracts.py", line 300, in setUp
    self._item_id = add_resp["result"]["id"]
  File ".../KrabEar/tests/test_swift_response_contracts.py", line 48, in _make_service
    return BackendService(store=store)
  File ".../KrabEar/backend/service.py", line 1103, in __init__
    self._dispatch_table: dict[...] = (
  File ".../KrabEar/backend/bookmarks.py", line 43, in __init__
    self._settings_get = settings_get or (lambda k, d: d)
  File ".../pathlib/__init__.py", line 1020, in mkdir
    if not exist_ok or not self.is_dir():
+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
```

Full thread dump at the moment of the timeout shows a genuinely stuck `stt-warmup` thread:

```
~~~~~~~~~~~~~~~~~~~~~~ Stack of stt-warmup ~~~~~~~~~~~~~~~~~~~~~~
  File ".../KrabEar/core/engine.py", line 487, in warmup
    mlx_whisper.transcribe(
  File ".../mlx_whisper/transcribe.py", line 297, in transcribe
    result: DecodingResult = decode_with_fallback(mel_segment)
  File ".../mlx_whisper/decoding.py", line 740, in decode
    result = DecodingTask(model, options).run(mel)
  File ".../mlx_whisper/decoding.py", line 650, in run
    tokens, sum_logprobs, no_speech_probs = self._main_loop(audio_features, tokens)
  File ".../mlx_whisper/decoding.py", line 607, in _main_loop
    next_tokens, next_completed, next_sum_logprobs, _ = _step(...)
  File ".../mlx_whisper/decoding.py", line 584, in _step
    logits = logit_filter.apply(logits, tokens)
  File ".../mlx_whisper/decoding.py", line 343, in apply
    tokens = tokens.tolist()
```

i.e. a `stt-warmup` daemon thread is stuck deep inside **real** `mlx_whisper.transcribe()` GPU inference (not mocked out for this test run) and never returns. Since `test_swift_response_contracts.py`'s `_make_service()` constructs plain `BackendService(store=store)` objects repeatedly without `.close()` in `tearDown` (see the many stacked `DiskSpaceMonitor`/`PurgeScheduler`/`export-scheduler`/`LLMHttpProbe` threads also visible in the dump — the known #1782 daemon-thread-leak class), many BackendService instances accumulate over the run, and eventually one of their STT warmup threads deadlocks/hangs indefinitely inside MLX inference, stalling the entire suite.

Likely root cause candidates:
1. `test_swift_response_contracts.py` needs `service.close()` in `tearDown` per the existing #1782 convention (`BackendService` daemon-thread teardown).
2. Real STT warmup (`core/engine.py:487`, `mlx_whisper.transcribe`) is running during unit tests instead of being mocked/disabled — this should not hit real MLX GPU inference in the test suite at all, and per CLAUDE.md's MLX thread-safety rules, concurrent unlocked MLX inference (from multiple accumulated warmup threads) is exactly the class of bug that causes stalls/SIGSEGV.

## Swift suite: PASS
```
Test Suite 'KrabEarAgentPackageTests.xctest' passed at 2026-07-01 14:29:04.397.
	 Executed 1090 tests, with 6 tests skipped and 0 failures (0 unexpected) in 59.476 (59.560) seconds
```

## Action needed
Investigate `test_swift_response_contracts.py` for missing `service.close()` in `tearDown`, and confirm STT warmup is properly mocked/disabled for tests using `BackendService` (should never reach real `mlx_whisper.transcribe`).

_Filed automatically by the scheduled `krab-ear-test-health` task._

**Comment by Pavua** [comment]
Дубликат ежедневного scheduled-task `krab-ear-test-health`.

Хронический локальный hang полного pytest tracked в #1909.
Последний осмысленный ubuntu-сигнал — #1919 (2026-08-13).

Закрыто волной W1 (2026-08-16). Генератор больше не должен открывать
`ci: test suite failure — DATE` на тот же hang.
